qa/tasks/rsync.py

```diff
@@ -0,0 +1,283 @@
+
+    runtime: Required Param for using rsync task sequentially. unit in seconds. default 0 sec.
+             Don't use this param during parallel execution. if used then finally block will be delayed
+
+    data_dir: Don't use this param if not rsyncing data/snap from workunit data
+              Optional Param. Specify the client like client.0/client.1
+              make sure cephfs mount and workunit task using this client according to scenario.
+              Default source directory will be source/subdir for other scenarios.
+
+    snapenable: Optional Param. value in boolean (True or False), default False(disabled).
+                using this option will enable cephfs snapshot.
+
+    waittime: Optional Param. units in seconds. default 5 sec.
+              During rsync iteration each loop will wait for specified seconds.
+
+    mountpoint: Optional param. default it will be first ceph-fuse client id.
+            eg: if there are 3 cephfs clients then for client.0 use 0, client.1 use 1
+            Setting this option will run rsync on this particular client, make sure cephfs mount available.
+
+    Examples:
+    rsync task parallel with another task
+    tasks:
+      - ceph:
+      - ceph-fuse:
+      - rsync:
+            waittime: 10
+            snapenable: True
+            filesize: 8
+      - cephfs_test_runner:
+
+    rsync data from workunit directory in same client.
+    tasks:
+      - ceph:
+      - ceph-fuse: [client.2]
+      - rsync:
+            waittime: 10
+            mountpoint: 2
+            data_dir: client.2
+      - workunit:
+            clients:
+            client.2:
+                - suites/iozone.sh
+
+    rsync task parallel with workunit but on different client
+    tasks:
+      - ceph:
+      - ceph-fuse:
+      - rsync:
+            waittime: 10
+            mountpoint: 1
+      - workunit:
+            clients:
+            client.2:
+                - suites/iozone.sh
+
+    rsync task sequentially based on runtime. default runtime is 0
+    tasks:
+      - ceph:
+      - ceph-fuse:
+      - rsync:
+            runtime: 120
+            mountpoint: 2
+            snapenable: True
+
+'''
+
+class RSync(Greenlet):
+
+    def __init__(self, ctx, config, logger):
+        super(RSync, self).__init__()
+
+        self.ctx = ctx
+
+        self.stopping = Event()
+        self.config = config
+        self.logger = logger
+
+        self.my_mnt = None
+        self.work_unit = False
+
+        self.file_size = self.config.get('filesize', 2)
+        self.wait_time = self.config.get('waittime', 5)
+        self.mount_point = self.config.get('mountpoint')
+
+        self.snap_enable = bool(self.config.get('snapenable', False))
+
+        self.data_dir = self.config.get('data_dir')
+
+        #Get CephFS mount client and mount object
+        assert len(self.ctx.mounts.items()) > 0, 'No mount available asserting rsync'
+
+        if len(self.ctx.mounts.items()):
```

Comment:
This is always true here. That's why it didn't assert.

Reply:
Ack. Updated.